pkg/client/client.go

```diff
@@ -1004,9 +1017,35 @@ func monitorKeepTerminatedEvent(
 					members,
 				)
 
-				tss.BroadcastRecoveryAddress(
+				if err != nil {
+					logger.Errorf(
+						"failed to retrieve member ids on keep [%s] termination: [%v]",
+						keep.ID().Hex(),
+						err,
+					)
+					return
+				}
+
+				var beneficiaryAddress string
+				_, err = btcutil.DecodeAddress(bitcoinConfig.BeneficiaryAddress, &chaincfg.TestNet3Params)
+				if err != nil {
+					beneficiaryAddress, err = recovery.DeriveAddress(bitcoinConfig.BeneficiaryAddress, 0)
```

Comment:
I'd love to see a comment in the code explaining why we look for an error of `DecodeAddress` and why in one case we're deriving the address and in the other, we're using the address straight from the config.